django/urls/resolvers.py

```diff
@@ -0,0 +1,293 @@
+from __future__ import unicode_literals
+
+from functools import update_wrapper
+
+from django.core.exceptions import ImproperlyConfigured
+from django.urls.exceptions import NoReverseMatch, Resolver404
+from django.utils import six
+from django.utils.decorators import available_attrs
+from django.utils.functional import cached_property
+from django.utils.module_loading import import_module
+
+from .utils import get_callable
+
+
+class ResolverMatch(object):
+    def __init__(self, func, args, kwargs, url_name=None, app_names=None, namespaces=None, decorators=None):
+        self.func = func
+        self.args = args
+        self.kwargs = kwargs
+        self.url_name = url_name
+        self.app_names = [x for x in app_names if x] if app_names else []
```

Comment:
add comment preceding to explain why a new list is being created for `app_names` and `namespaces`? Is it because they're lazy, or working on an explicit copy'd value is needed, or because they're an iterable object but we need to act specifically on a list?